src/NumericalAlgorithms/Spectral/SwshCollocation.cpp

```diff
@@ -0,0 +1,99 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include <cmath>
+
+#include "ErrorHandling/Assert.hpp"
+#include "NumericalAlgorithms/Spectral/ComplexDataView.hpp"
+#include "NumericalAlgorithms/Spectral/SwshCollocation.hpp"
+
+// IWYU pragma: no_include <sharp_geomhelpers.h>
+// IWYU pragma: no_include <sharp_lowlevel.h>
+
+namespace Spectral {
+namespace Swsh {
+
+// @{
+/// (In)Equivalence checks both the object and index for the iterator
+bool operator==(typename Collocation<ComplexRepresentation::RealsThenImags>::
```

Comment:
since the definitions are the same, why not just write it once as a template and then instantiate it twice?